src/openrct2-ui/windows/ScenarioSelect.cpp

```diff
@@ -125,6 +126,38 @@ namespace OpenRCT2::Ui::Windows
         const ScenarioIndexEntry* _highlightedScenario = nullptr;
         ParkPreview _preview;
         BackgroundWorker::Job _previewLoadJob;
+        bool _initialLoadCommand = true;
+
+        void ReportProgress(uint8_t progress)
+        {
+            if (_initialLoadCommand)
+                return;
+
+            if (progress == 0)
+                GetContext()->OpenProgress(STR_REBUILDING_SCENARIO_INDEX);
+
+            GetContext()->SetProgress(progress, 100, STR_STRING_M_PERCENT);
+
+            if (progress == 100)
+                GetContext()->CloseProgress();
+        }
+
+        void LoadScenarioList()
+        {
+            try
+            {
+                ReportProgress(0);
+                ScenarioRepositoryScan();
+                ReportProgress(100);
```

Comment:
This just jumps from 0 to 100, rather than providing real feedback.

Reply:
`ScenarioRepositoryScan` leads to `_scenarioRepository->Scan`, which should take care of updating progress, similar to how it's used in `Context::InitialiseRepositories`.

Reply:
I see.